.github/workflows/static-code-analysis.yml

```diff
@@ -39,3 +39,23 @@ jobs:
 
     - name: Perform CodeQL Analysis
       uses: github/codeql-action/analyze@v1
+
+  cppcheck:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
```

Comment:
You want `pull-request: write`

Reply:
Thanks for catching that. I'm not having my best week

Reply:
In the end, you were right when you used pull_request_target. When the pull_request event is run from a forked repository, you can never get a write permission. In this case, I'd suggest to move it to a standalone workflow to avoid having multiple entries (cppcheck (pull_request) - skipped, cppcheck (pull_request_target), codeql (pull_request), codeql (pull_request_target - skipped).